arch/Mips/MipsDisassembler.c

```diff
@@ -1374,9 +1374,9 @@ static DecodeStatus readInstruction16(const uint8_t *Bytes, size_t BytesLen,
 	}
 
 	if (IsBigEndian) {
-		*Insn = (Bytes[0] << 8) | Bytes[1];
+		*Insn = ((uint64_t)Bytes[0] << 8) | (uint64_t)Bytes[1];
 	} else {
-		*Insn = (Bytes[1] << 8) | Bytes[0];
+		*Insn = ((uint64_t)Bytes[1] << 8) | (uint64_t)Bytes[0];
```

Comment:
Please move these and the one below into new `readBytes16u64()`, `readBytes32u64()` functions in `utils.c`. They are universally useful.

Reply:
You can also move the `NanoMips` byte ordering into it. We could expose those things in the future in the API.